### PR TITLE
Remove references to admin:admin

### PR DESCRIPTION
**Pull request by vibrantvarun**
### Description
Remove default password of admin. Recently, security repo has mandated that there will no admin:admin password from 2.12. Therefore, removing default admin password from developer guide in this PR.

### Issues
Part of [https://github.com/opensearch-project/security/issues/3624](https://github.com/opensearch-project/security/issues/3624)
Resolves: 
[https://github.com/opensearch-project/neural-search/issues/550](https://github.com/opensearch-project/neural-search/issues/550)

### Check List
- [x] New functionality includes testing.
    - [x] All tests pass
- [x] New functionality has been documented.
    - [x] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vibrantvarun** [comment]
The security tests are expected to fail as the support for custom password is enabled from 2.12

File: build.gradle [hunk 1]
```diff
@@ -85,9 +86,9 @@ ext {
         cluster.getNodes().forEach { node ->
             var creds = node.getCredentials()
             if (creds.isEmpty()) {
-                creds.add(Map.of('username', 'admin', 'password', 'admin'))
+                creds.add(Map.of('username', 'admin', 'password', System.getProperty("password")))
```
Comment (DarshitChanpura) [inline build.gradle:89]: his should be: System.getProperty("OPENSEARCH_INITIAL_ADMIN_PASSWORD")

**Review by DarshitChanpura — CHANGES_REQUESTED** [review]
@vibrantvarun ty for this PR. I've suggested a change I think is needed to fetch the correct password.

File: build.gradle (hunk 1 above)
Comment (vibrantvarun) [reply build.gradle:89]: line 306 will set as default password OPENSEARCH_INITIAL_ADMIN_PASSWORD if the password is not given by user .

Comment (DarshitChanpura) [reply build.gradle:89]: the tests should not fail for main, as the logic is already in place unless main is not being used for test.

There should be version check in place that uses `admin` password for versions <2.12

Comment (vibrantvarun) [reply build.gradle:89]: Let me know If still I have to make it OPENSEARCH_INITIAL_ADMIN_PASSWORD

Comment (DarshitChanpura) [reply build.gradle:89]: > line 306 will set as default password OPENSEARCH_INITIAL_ADMIN_PASSWORD if the password is not given by user .

qq, when will the property `password` be set?

Comment (vibrantvarun) [reply build.gradle:89]: I have tested on my local machine that if I do not set password then it will set OPENSEARCH_INITIAL_ADMIN_PASSWORD the default password

Comment (vibrantvarun) [reply build.gradle:89]: When we trigger ./gradlew integTest -Dsecurity.enabled=true then it will first set password [https://github.com/opensearch-project/neural-search/blob/98e5534929efb054e7b5d4382af46b0626ef1f50/build.gradle#L305](https://github.com/opensearch-project/neural-search/blob/98e5534929efb054e7b5d4382af46b0626ef1f50/build.gradle#L305)

Comment (vibrantvarun) [reply build.gradle:89]: I think I figured out the change. Reverting it to OPENSEARCH_INITIAL_ADMIN_PASSWORD

Comment (DarshitChanpura) [reply build.gradle:89]: ty.. in that case.. we can add a version based password setting on line 306 . Thoughts?

Comment (vibrantvarun) [reply build.gradle]: I think I understood the point you are trying to make. Therefore set the password as OPENSEARCH_INITIAL_ADMIN_PASSWORD now

File: build.gradle [hunk 2]
```diff
@@ -62,6 +62,7 @@ def opensearch_tmp_dir = rootProject.file('build/private/opensearch_tmp').absolu
 opensearch_tmp_dir.mkdirs()
 
 ext {
+    System.setProperty("OPENSEARCH_INITIAL_ADMIN_PASSWORD", "myStrongPassword123!")
```
Comment (DarshitChanpura) [inline build.gradle:65]: This would not work for versions <2.12 because at line 89, 91, 306 where its actually using the password, the expected password is `admin` for <2.12 but it will use `myStrongPassword123!` instead

Comment (vibrantvarun) [reply build.gradle:65]: So shall I put the version check?

Comment (DarshitChanpura) [reply build.gradle:65]: yes..that would resolve any issues with password for <2.12

File: build.gradle [hunk 3]
```diff
@@ -302,7 +312,7 @@ integTest {
         // If security is enabled, set is_https/user/password defaults
         is_https = is_https == null ? "true" : is_https
         user = user == null ? "admin" : user
-        password = password == null ? "admin" : password
+        password = password == null ? System.getProperty("OPENSEARCH_INITIAL_ADMIN_PASSWORD") : password
```
[the remark this answers is not in the document]
Comment (vibrantvarun) [reply build.gradle:315]: No, in backwards_compatibility_tests_workflow.yml

security tests are called by below command

```
          su `id -un 1000` -c "whoami && java -version && ./gradlew integTest -Dsecurity.enabled=true"
```

Here if you see I have not set any password here

**Comment by DarshitChanpura** [comment]
This PR is not needed since the neural-search `integTest` task doesn't use `install_demo_configuration.sh` to setup the cluster. Instead it installs the plugin and sets the [opensearch security configuration manually](https://github.com/opensearch-project/neural-search/blob/bbc76337f2f54c64375b7fa33313e921f7210266/build.gradle#L78-L141). Closing this PR now. Feel free to re-open if I missed something.

File: DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -137,12 +137,11 @@ Additionally, it is also possible to run a cluster with security enabled:
 ./gradlew run -Dsecurity.enabled=true
 ```
 
-By default, if `-Dsecurity.enabled=true` is passed the following defaults will be used: `https=true`, `user=admin` and
-`password=admin`.
+By default, if `-Dsecurity.enabled=true` is passed the following defaults will be used: `https=true`, `user=admin`. There is no default password and it is to be set as `password=<admin-password>`.
 
-Then, to access the cluster, we can run
+Then, to access the cluster, we can run the below command. <admin-password> parameter is the password set by admin.
```
Comment (DarshitChanpura) [inline DEVELOPER_GUIDE.md:142]: nit: 
```
To connect to the cluster, use the following command. Remember to replace `<admin-password>` with the password you chose when setting up the admin user.
```

**Review by DarshitChanpura — APPROVED** [review]
LGTM!

File: gradle/formatting.gradle [hunk 5]
```diff
@@ -19,7 +19,7 @@ allprojects {
             }
         }
         format 'misc', {
-            target '*.md', '**/*.gradle', '**/*.json', '**/*.yaml', '**/*.yml', '**/*.svg'
+            target '*.md', '**/*.gradle', '**/*.json', '**/*.yaml', '**/*.yml', '**/*.svg', '**/*.properties'
```
Comment (martin-gaievski) [inline gradle/formatting.gradle:22]: is this related to PR?

**Review by martin-gaievski — APPROVED** [review]

**Review by junqiu-lei — APPROVED** [review]
